packages/creator/src/creator.ts

```diff
@@ -35,228 +53,90 @@ import * as createItemTemplate from "./createItemTemplate";
  */
 export function createSolution(
   sourceId: string,
-  authentication: common.UserSession,
-  options?: common.ICreateSolutionOptions
+  authentication: UserSession,
+  options?: ICreateSolutionOptions
 ): Promise<string> {
-  return new Promise((resolve, reject) => {
-    const createOptions: common.ICreateSolutionOptions = options || {};
-    if (createOptions.progressCallback) {
-      createOptions.progressCallback(1); // let the caller know that we've started
-    }
-
-    // Get group information
-    // tslint:disable-next-line: no-floating-promises
-    Promise.all([
-      common.getGroupBase(sourceId, authentication),
-      common.getGroupContents(sourceId, authentication)
-    ])
-      .then(
-        responses => {
-          const [groupInfo, groupItems] = responses;
-          if (createOptions.progressCallback) {
-            createOptions.progressCallback(15);
-          }
-
-          // Create a solution from the group's contents, using the group's information as defaults for the solution item
-          createOptions.title = createOptions.title ?? groupInfo.title;
-          createOptions.snippet = createOptions.snippet ?? groupInfo.snippet;
-          createOptions.description =
-            createOptions.description ?? groupInfo.description;
-          createOptions.tags = createOptions.tags ?? groupInfo.tags;
-
-          /* istanbul ignore else*/ if (
-            !createOptions.thumbnailurl &&
-            groupInfo.thumbnail
-          ) {
-            // Copy the group's thumbnail to the new item
-            // createOptions.thumbnail needs to be a full URL
-            createOptions.thumbnailurl = common.generateSourceThumbnailUrl(
-              authentication.portal,
-              sourceId,
-              groupInfo.thumbnail,
-              true
-            );
-          }
-
-          // Create a solution with the group contents
-          return groupItems;
-        },
+  let createOptions: ICreateSolutionOptions = options || {};
+  const progressCb = createOptions.progressCallback || noOp;
```

Comment:
We could consider doing something like this in a number of places as it removes a whole bunch of checks to see if the callback exists... it either does, or it's a `noOp` function and either case, we just call it as needed